### DISCUSSION TITLE
Github copilot stops responding

### DISCUSSION THREAD

**Discussion by zachariasnuernberger** [Genuitec/Copilot4Eclipse-community#15] [Help]
It happens several times a day that the copilot does not answer my questions and the automatic completion no longer makes any suggestions.

The firewall was checked but no error was found

My programming language is abap

Copilot Session Status
Connection: OK
STATUS: Copilot4Eclipse ready
Copilot API Version: 1.180.0

**Comment by wayneparrott** [comment]
Thanks for sharing. Does this non-responsive condition clear up on its own or do you have to take an action such as restarting Eclipse to restore completions?

**Reply by zachariasnuernberger** [reply]
No, sometimes it works and sometimes it doesn't - a restart does nothing
My work college has exactly the same problem

**Comment by wayneparrott** [comment]
Questions:

1. Approximately how long does it take from when you start Eclipse until Copilot4Eclipse becomes unresponsive?

1a.  Is it about the same time period for each incident?

A few things to help us get a better picture of your situation:

1. Please check the Eclipse log (Eclipse WIndows menu > Show View > Error Log) for useful error entries and share here if available.

2. Turn on LSP Logging and look for useful output, (Eclipse Windows menu > Preferences > Language Servers > Log, enable Copilot4Eclipse language server). Note we do not recommend enabling LSP logging except for when debugging otherwise you'll have a ton of logging overhead.

3. The next time Copilot is unresponsive try logging out and then logging back in. This is different from restarting Eclipse. Share the results.

There are times when we have seen GitHub Copilot services are offline or unresponsive. In such cases we just have to wait from them to be back online. But I seldom see such issue in recent months. Also I have encountered some situations when I was testing completions in an abap editor that the generated text was not very helpful.

Out of curiosity, what level is your GitHub Copilot subscriptions (individual, biz, enterprise)?

W
